Dockerfile

```diff
@@ -82,6 +82,32 @@ ARG BINARY_URL
 RUN curl -Lo /bin/$PROJECT_BIN $BINARY_URL
 RUN chmod +x /bin/$PROJECT_BIN
 
+FROM gcc:12 AS zstd_build
```

Comment:
isn't it easier to `apt -y install zstd` ? :)

Reply:
Good question (I forgot to mention it in description)

I have checked the easy way first (install via package manager)
But [the version available to debian buster is `1.3.8`](https://packages.debian.org/buster/zstd) which is [released on 28 Dec 2018](https://github.com/facebook/zstd/releases/tag/v1.3.8)

[1.5.0](https://github.com/facebook/zstd/releases/tag/v1.5.0) (and 1.5.x) is released with large performance improvement (claimed in released description) and there might be options that are only introduced in recent versions

Regarding use of gcc instead of clang
I read this: https://github.com/facebook/zstd/issues/2695 (which also contains a linked issue with benchmark)

I can't say which compiler or version is better for performance (especially it might depend on the compression level and/or other options to be used) so I just go with latest `gcc`
Let me know if anyone got better idea at this

Reply:
`buster` is the old stable..  `bullseye` (current stable) has zstd `1.4.8`.

I think a quick measurement compressing some of the `*.ldb` files using the zstd `1.3.8` and zstd `1.4.8` is worth it.
As I don't think it is worth compiling the entire `zstd` if it only wins less than 3% of performance (either the compression ratio or the cpu usage), especially if we can get it naturally with time (with newer debian releases).

Neither I like the idea of it curl'ing the `https://bootstrap.pypa.io/get-pip.py` as one day it can get hacked / DNS hijacked / etc.. maybe I'm just too paranoid :man_shrugging:

Reply:
Which is why **only ** the built binary is coped via `COPY --from=zstd_build /usr/local/bin/zstd /bin/` later
The stuff installed in build stage `zstd_build` **won't** be available to final image `omnibus`

Reply:
> Which is why **only ** the built binary is coped via `COPY --from=zstd_build /usr/local/bin/zstd /bin/` later The stuff installed in build stage `zstd_build` **won't** be available to final image `omnibus`

Yeah, I know that. 
I meant the other way around - infection through the added code into the zstd itself.
But, again, paranoia--.